mobile-app/app/screens/AppNavigator/screens/Balances/screens/AddressBookScreen.tsx

```diff
@@ -33,6 +38,28 @@ export function AddressBookScreen ({ route, navigation }: Props): JSX.Element {
   const userPreferences = useSelector((state: RootState) => state.userPreferences)
   const addressBook = userPreferences.addressBook
   const [isEditing, setIsEditing] = useState(false)
+  const { getAddressUrl } = useDeFiScanContext()
+
+  // Search
+  const [showSearchInput, setShowSearchInput] = useState(false)
+  const [searchString, setSearchString] = useState('')
+  const filterAddress = debounce((searchString: string): void => {
+    if (searchString !== undefined && searchString.trim().length > 0) {
+      const addressBookList: string[] = []
+
+      for (const address in addressBook) {
+        if (address.includes(searchString.trim().toLowerCase()) || addressBook[address].label.toLowerCase().includes(searchString.trim().toLowerCase())) {
```

Comment:
Might want to check if this exist = addressBook[address]

Reply:
added optional chaining when accessing nested label